polaris/dataset/_base.py

```diff
@@ -215,6 +232,18 @@ def dtypes(self) -> dict[str, np.dtype]:
         """Return the dtype for each of the columns for the dataset"""
         raise NotImplementedError
 
+    def should_verify_checksum(self, strategy: ChecksumStrategy) -> bool:
```

Comment:
**Comment**: Verifying the checksum (i.e. recomputing the checksum and comparing it against the provided checksum), only makes sense for the V1 implementation because the checksum for the V2 is not deterministic.